localstack-core/localstack/aws/client.py

```diff
@@ -221,3 +272,74 @@ def raise_service_exception(response: Response, parsed_response: Dict) -> None:
     """
     if service_exception := parse_service_exception(response, parsed_response):
         raise service_exception
+
+
+@patch(Endpoint.create_request)
+def _create_and_enrich_request(
+    fn, self: Endpoint, params: dict, operation_model: OperationModel = None
+):
+    """
+    Patch that adds the botocore operation model and request parameters to a newly created AWSPreparedRequest, which normaly only holds low-level HTTP request information. This
+    """
+    request: awsrequest.AWSPreparedRequest = fn(self, params, operation_model)
+
+    request.params = params
+    request.operation_model = operation_model
+
+    return request
+
+
+class GatewayShortCircuit:
+    gateway: Gateway
+
+    def __init__(self, gateway: Gateway):
+        self.gateway = gateway
+        self._internal_url = get_service_endpoint()
+
+    def __call__(
+        self, event_name: str, request: awsrequest.AWSPreparedRequest, **kwargs
+    ) -> awsrequest.AWSResponse | None:
+        # TODO: we sometimes overrides the endpoint_url to direct it to DynamoDBLocal directly
+        # if the default endpoint_url is not in the request, just skips the in-memory forwarding
+        if self._internal_url not in request.url:
+            return
+
+        # extract extra data from enriched AWSPreparedRequest
+        params = request.params
+        operation: OperationModel = request.operation_model
+
+        # create request
+        context = RequestContext()
+        context.request = create_http_request(request)
+
+        # TODO: just a hacky thing to unblock the service model being set to `sqs-query` blocking for now
+        # this is using the same services as `localstack.aws.protocol.service_router.resolve_conflicts`, maybe
+        # consolidate. `docdb` and `neptune` uses the RDS API and service.
+        if operation.service_model.service_name not in {
+            "sqs-query",
+            "docdb",
+            "neptune",
+            "timestream-write",
+        }:
+            context.service = operation.service_model
+
+        context.operation = operation
+        context.service_request = params["body"]
```

Comment:
So the request parameters are always stored in "body"? Interesting :) If I undestand this correctly, this should let us avoid service detection and service request parsing, right?

Reply:
So it seems 😄 it exactly does, yes! It skips the service detection, but as the hacky way above demonstrate, it can lead to a few issues so I don't set it there on purpose for those services.